linters/namedfields/namedfields.go

```diff
@@ -0,0 +1,107 @@
+package namedfields
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"reflect"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+const defaultFieldThreshold = 5 // Require named fields for structs with more than 5 fields
+
+var Analyzer = &analysis.Analyzer{
+	Name:       "namedfields",
+	Doc:        "check that struct literals with many fields use named field initialization",
+	Run:        func(p *analysis.Pass) (interface{}, error) { return run(false, p) },
+	ResultType: reflect.TypeOf(Result{}),
+}
+
+var analyzerForTests = &analysis.Analyzer{
+	Name:       "testnamedfields",
+	Doc:        "check that struct literals with many fields use named field initialization (for tests)",
+	Run:        func(p *analysis.Pass) (interface{}, error) { return run(true, p) },
+	ResultType: reflect.TypeOf(Result{}),
+}
+
+type namedFieldsError struct {
+	Pos     token.Pos
+	Message string
+}
+
+type Result struct {
+	Errors []namedFieldsError
+}
+
+func run(dryRun bool, pass *analysis.Pass) (interface{}, error) {
+	var ret Result
+	threshold := defaultFieldThreshold
```

Comment:
since there is no other way we can customize the analyzer behavior than changing the constant, maybe we could remove the `threshold` variable and rename `defaultFieldThreshold` to just `fieldThreshold`?